ccm-fetch fails claiming illegal characters in kerberos principal

Using ccm 16.10.0-1.
Output of component ccm :
`[ERROR] failed to ccm-fetch with new config: Uncaught exception!!! Calling stack is:
  LC::Exception::throw_error called at /usr/lib/perl/EDG/WP4/CCM/Fetch/Config.pm line 250
  EDG::WP4::CCM::Fetch::Config::setPrincipal called at /usr/sbin/ccm-fetch line 280
*** Invalid characters in principal: host/test-krb.wn.iihe.ac.be@WN.IIHE.AC.BE
`
The regular expression that is used to check the principal doesn't look to me correct :
`^[\w.-\/@]+$`
For me, it should be something like :
`^[\w\-\.\/@]+$`
because hyphen and dot have special meanings in square brackets.

--- a/src/main/perl/Fetch/Config.pm
+++ b/src/main/perl/Fetch/Config.pm
@@ -225,7 +225,7 @@ sub setPrincipal
 {
     my ($self, $val) = @_;
     throw_error("Invalid characters in principal: $val")
-        unless ($val =~ m/^[\w.-\/@]+$/);
+        unless ($val =~ m/^[\w.\-\/@]+$/);
     $self->{PRINCIPAL} = $val;
     return SUCCESS;
 }

--- a/src/test/perl/fetch-config.t
+++ b/src/test/perl/fetch-config.t
@@ -2,10 +2,29 @@ use strict;
 use warnings;
 use Test::More;
 
+use LC::Exception;
 use EDG::WP4::CCM::Fetch::Config qw(NOQUATTOR NOQUATTOR_EXITCODE NOQUATTOR_FORCE);
 
 is(NOQUATTOR, "/etc/noquattor", "Exported NOQUATTOR");
 is(NOQUATTOR_EXITCODE, "3", "Exported NOQUATTOR_EXITCODE");
 is(NOQUATTOR_FORCE, "force-quattor", "Exported NOQUATTOR_FORCE");
+
+my $ec = LC::Exception::Context->new->will_store_errors;
+# Use a plain hash as fake instance
+# This is ok to test the exception throwing
+my $inst = {};
+
+my $prp = 'service/test-server.domain@realm.something';
+ok(EDG::WP4::CCM::Fetch::Config::setPrincipal($inst, $prp),
+   "setPrincipal ok");
+is($inst->{PRINCIPAL}, $prp, "setPrincipal sets PRINCIPAL attribute");
+ok(! $ec->error, "No error thrown by valid principal");
+
+# This tests that the backslash is not a valid char in the regexp,
+# and the the . is a ., not just any char
+EDG::WP4::CCM::Fetch::Config::setPrincipal($inst, 'random/garbage\with@backslash');
+ok($ec->error, "Error thrown by invalid principal");
+like($ec->error->text, qr{}, "expected exception message thrown by invalid principal");
+$ec->ignore_error;
 
 done_testing;

